pootle/apps/pootle_store/views.py

```diff
@@ -407,102 +410,122 @@ def save_comment(request, unit):
-        snplurals = len(unit.source.strings)
-    else:
-        snplurals = None
-
-    form_class = unit_form_factory(language, snplurals, request)
-    form = form_class(instance=unit, request=request)
-    comment_form_class = unit_comment_form_factory(language)
-    comment_form = comment_form_class({}, instance=unit, request=request)
-
-    store = unit.store
-    directory = store.parent
-    user = request.user
-    project = translation_project.project
-
-    alt_src_langs = get_alt_src_langs(request, user, translation_project)
-    altsrcs = find_altsrcs(unit, alt_src_langs, store=store, project=project)
-    source_language = translation_project.project.source_language
-    sources = {
-        unit.store.translation_project.language.code: unit.target_f.strings
-        for unit in altsrcs
-    }
-    sources[source_language.code] = unit.source_f.strings
-
-    priority = None
-
-    if 'virtualfolder' in settings.INSTALLED_APPS:
-        vfolder_pk = request.GET.get('vfolder', '')
-
-        if vfolder_pk:
-            from virtualfolder.models import VirtualFolder
-
-            try:
-                # If we are translating a virtual folder, then display its
-                # priority.
-                # Note that the passed virtual folder pk might be invalid.
-                priority = VirtualFolder.objects.get(pk=vfolder_pk).priority
-            except VirtualFolder.DoesNotExist:
-                pass
-
-        if priority is None:
-            # Retrieve the unit top priority, if any. This can happen if we are
-            # not in a virtual folder or if the passed virtual folder pk is
-            # invalid.
-            priority = unit.vfolders.aggregate(
-                priority=Max('priority')
-            )['priority']
-
-    template_vars = {
-        'unit': unit,
-        'form': form,
-        'comment_form': comment_form,
-        'priority': priority,
-        'store': store,
-        'directory': directory,
-        'user': user,
-        'project': project,
-        'language': language,
-        'source_language': source_language,
-        'cantranslate': check_user_permission(user, "translate", directory),
-        'cansuggest': check_user_permission(user, "suggest", directory),
-        'canreview': check_user_permission(user, "review", directory),
-        'is_admin': check_user_permission(user, 'administrate', directory),
-        'altsrcs': altsrcs,
-    }
-
-    if translation_project.project.is_terminology or store.is_terminology:
-        t = loader.get_template('editor/units/term_edit.html')
-    else:
-        t = loader.get_template('editor/units/edit.html')
-    c = RequestContext(request, template_vars)
-
-    json.update({
-        'editor': t.render(c),
-        'tm_suggestions': unit.get_tm_suggestions(),
-        'is_obsolete': unit.isobsolete(),
-        'sources': sources,
-    })
-
-    return JsonResponse(json)
+class UnitEditJSON(PootleJSON):
+
+    model = Unit
+    pk_url_kwarg = "uid"
```

Comment:
see here to understand their use https://docs.djangoproject.com/en/1.9/ref/class-based-views/generic-display/#detailview